### PR TITLE
Add an option to select a serviceAccount

### LINKED ISSUE #244: [Feature] Allow setting ServiceAccountName / don't use "default" ServiceAccount
**Is your feature request related to a problem? Please describe.**
When applying security policies like "no default service account name", currently the only way to do this is by creating a galera cluster. 

https://github.com/mariadb-operator/mariadb-operator/blob/b25c694a80d1064de120d623fede25824af9eda5/pkg/builder/statefulset_builder.go#L180-L187

**Describe the solution you'd like**
I would like to either set the name manual, or just set a "random" name in `buildStsServiceAccountName` when not galera.

**Describe alternatives you've considered**
Alternative is deploying galera cluster. but you might not want to do that.

**Additional context**
-

**Environment details**:
- Kubernetes version: 1.26
- mariadb-operator version: latest (0.20.0)
- Install method: helm
- Install flavour: custom (i guess)

### PR DESCRIPTION
**Pull request by Hoega**
Closes #244  and #333 
Instead of using the default service account, a new service account with the mariadb name will be used:

`k get po mariadb-min-0 -o yaml`
```
     volumeMounts:
    - mountPath: /var/lib/mysql
      name: storage
    - mountPath: /etc/mysql/conf.d
      name: config
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-n2s4z
      readOnly: true
  automountServiceAccountToken: true
  serviceAccount: mariadb-min
  serviceAccountName: mariadb-min
```

You can also now specify a service account with a new option serviceAccountName:
```
apiVersion: mariadb.mmontes.io/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-min
spec:
  serviceAccountName: test123
```

If the service account does not exist, it will be created:
`  k get serviceaccounts `
```
NAME      SECRETS   AGE
default   0         49m
test123   0         114s
```

### CODE REVIEW

File: pkg/builder/statefulset_builder.go [hunk 1]
```diff
@@ -157,12 +158,13 @@ func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.Per
...
 func buildStsServiceAccountName(mariadb *mariadbv1alpha1.MariaDB) (autoMount *bool, serviceAccount string) {
-	if mariadb.Galera().Enabled {
-		mount := false
-		autoMount = &mount
-		serviceAccount = mariadb.Name
+	if mariadb.Spec.ServiceAccountName != nil {
+		if mariadb.Spec.Galera.Enabled {
+			return ptr.To(false), *mariadb.Spec.ServiceAccountName
+		}
+		return ptr.To(true), *mariadb.Spec.ServiceAccountName
```
Comment (Hoega) [inline pkg/builder/statefulset_builder.go:165]: Do we need the automount = true? The serviceaccount token was mounted when the default SA was used, is it used?

Comment (mmontes11) [reply pkg/builder/statefulset_builder.go:165]: We can set it to false in the non Galera clusters, as we are not currently using it. For Galera, we are explicitly mounting it here:
https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/builder/statefulset_builder.go#L197

Juding by https://github.com/mariadb-operator/mariadb-operator/issues/244 and https://github.com/mariadb-operator/mariadb-operator/issues/333, this issue is about be compliant with security policies, which in general are against the default `ServiceAccount`.

File: pkg/builder/statefulset_builder.go [hunk 2]
```diff
@@ -157,12 +158,13 @@ func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.Per
 }
 
 func buildStsServiceAccountName(mariadb *mariadbv1alpha1.MariaDB) (autoMount *bool, serviceAccount string) {
```
Comment (mmontes11) [inline pkg/builder/statefulset_builder.go:160]:
```suggestion
func buildStsServiceAccountName(mariadb *mariadbv1alpha1.MariaDB) (serviceAccount string) {
```
We no longer need the `autoMount` flag, we can hardcode this to `false` in the builder.

Comment (mmontes11) [reply pkg/builder/statefulset_builder.go:160]: For Galera, we are explicitly mounting it here:
https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/builder/statefulset_builder.go#L197

**Review by mmontes11 — CHANGES_REQUESTED** [review]
Thanks for your PR @Hoega ! As always, very much appreciated 🙏🏻 

I've left a couple of suggestions, let me know what you think

**Review by mmontes11 — APPROVED** [review]
LGTM! 👍🏻 Thanks for your contribution! 🦭

**Pull request merged**